cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/util/AwsIamUtil.java

```diff
@@ -0,0 +1,263 @@
+package com.sequenceiq.cloudbreak.cloud.aws.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import com.amazonaws.auth.policy.Action;
+import com.amazonaws.auth.policy.Policy;
+import com.amazonaws.auth.policy.Resource;
+import com.amazonaws.auth.policy.Statement;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.model.EvaluationResult;
+import com.amazonaws.services.identitymanagement.model.GetInstanceProfileRequest;
+import com.amazonaws.services.identitymanagement.model.GetRoleRequest;
+import com.amazonaws.services.identitymanagement.model.InstanceProfile;
+import com.amazonaws.services.identitymanagement.model.NoSuchEntityException;
+import com.amazonaws.services.identitymanagement.model.Role;
+import com.amazonaws.services.identitymanagement.model.ServiceFailureException;
+import com.amazonaws.services.identitymanagement.model.SimulatePrincipalPolicyRequest;
+import com.amazonaws.services.identitymanagement.model.SimulatePrincipalPolicyResult;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.cloudbreak.validation.ValidationResult.ValidationResultBuilder;
+
+public class AwsIamUtil {
```

Comment:
Why is this a util, why not AwsIamService?

Reply:
Because I definitely missed that one when I went through in a different package than I looked. Fixing it now.

Reply:
I definitely broke something with the latest push. The class aren't fully injected during the tests somehow. I end up with NPE when trying to use the `AwsIamService`

Reply:
Attila helped me out explaining how `@Spy` was needed since the nested injection doesn't work in the tests.